app/assets/javascripts/puchases.coffee

```diff
@@ -0,0 +1,140 @@
+$(document).on "keyup keypress", (event) ->
+  keyCode = event.keyCode || event.which
+  if keyCode == 13
+    event.preventDefault()
+    return false
+
+populateItems = (category_id, element) ->
+  id = parseInt category_id
+  for category in data.categories
+    if category.id is id
+      currentCategory = category
+  element.html tmpl("purchases-item-options-template", currentCategory)
+
+addPurchaseRow = (purchaseDetails) ->
+  # Add an empty row
+  $("#purchase-table tbody").append tmpl("purchases-new-purchase-template", {})
+  return unless purchaseDetails
+  # Populate that row if there are purchaseDetails
+  row = $("#purchase-table tbody tr:last")
+  category = row.find(".category")
+  item = row.find(".item")
+  quantity = row.find(".quantity")
+  variance = row.vind(".variance")
+
+  category.val purchaseDetails.category_id
+  category.trigger "change"
+  item.val purchaseDetails.item_id
+  item.trigger "change"
+  quantity.val purchaseDetails.quantity
+  variance.val purchaseDetails.variance
+
+expose "addPurchaseRows", ->
+  $ ->
```

Comment:
@tekHudson , i don't understand the two lines above. what does `expose` do, is it a built-in function or something defined in the code somewhere?

The next line with the standalone `$` is also confusing, is that an unused parameter for the function being defined?

Reply:
Sorry, I found the `expose.coffee` file, I know what that part is doing, now.

I'm still confused about line 33